swarm/pss/prox_test.go

```diff
@@ -212,24 +227,24 @@ func (d *testData) init(msgCount int) {
 // nodes Y and Z will be considered required recipients of the msg,
 // whereas nodes X, Y and Z will be allowed recipients.
 func TestProxNetwork(t *testing.T) {
-	t.Run("16/16", testProxNetwork)
+	t.Run("16/16/15", testProxNetwork)
 }
 
 // params in run name: nodes/msgs
 func TestProxNetworkLong(t *testing.T) {
 	if !*longrunning {
 		t.Skip("run with --longrunning flag to run extensive network tests")
 	}
-	t.Run("8/100", testProxNetwork)
-	t.Run("16/100", testProxNetwork)
-	t.Run("32/100", testProxNetwork)
-	t.Run("64/100", testProxNetwork)
-	t.Run("128/100", testProxNetwork)
+	t.Run("8/100/30", testProxNetwork)
```

Comment:
I would start using here a table driven test! I know we do string splitting magic in our code base in some tests. I don't think it's a good idea.

Just focusing here: I'm lost in finding the meaning of the 3 values. I guess we somewhere split the test name. I did not find that.

What about?
```
	tests := []struct {
		a,b,c, int
	}{
		// TODO: Add test cases.
	}
	for _, tt := range tests {
                fmt.Sprintf("%v/%v/%v", t.a,t.b, t.c)
		t.Run(name, func(t *testing.T) {
			testProxNetwork(t.a, t.b, t.c)
		})
	}
```

Reply:
Please make an issue and it will be a separate PR. @gluk256  ?

Reply:
- We are extending the already complex and dirty implementation in this PR by adding further functionality.
- I think the suggested change can be done in less than half an hour and would highly improve the readability.
=> I recommend to do the change as part of this PR. I just don't agree with adding more complexity when complexity could be also reduced achieving the same functionality.


(Fixing broken windows and refactoring should be part of our normal workflow if we want to continuously improve our code.)

Reply:
@frncmx 

> complex and dirty implementation

Well, your example isn't more readable to me. Sorry. Maybe it's a matter of taste.

I prefer to keep PRs focused to the point. This PR wasn't "fix readability of very recently added proximity test." If indeed that should be done, put it as another PR. Then that PR could be done in half an hour. Or we could add half an hour more and make it "use table tests instead of string parsing in tests." and make sure they all work.

Let's have room for more than one way of doing things, ok?

Reply:
I'll let go. I've created a separate issue: https://github.com/ethersphere/go-ethereum/issues/1308

> Well, your example isn't more readable to me. Sorry. Maybe it's a matter of taste.
@nolash  I feel a little bit uncomfortable around this. I think there is a clear canonical Go way here what provided readability and type safety. While seeing just "16/16/20" does not tell me what those parameters do. I've tried to make a nice description of the issue. I just only ask you read my questions in the "Problem" section and think about that. That's my last try.

Note: we talked and agreed with @gluk256 he will do the refactor. Feel free to mark the conversation resolved.